docs/terminal/shell-integration.md

```diff
@@ -91,6 +91,35 @@ When using shell integration, it has a "quality" associated with it that declare
 
 To view the shell integration quality, hover the terminal tab. Optionally, select **Show Details** on the hover to view more detailed information.
 
+## IntelliSense (Preview)
+
+### Background
+
+IntelliSense, powered by shell integration `setting(terminal.integrated.shellIntegration.enable)`, can be enabled with `setting(terminal.integrated.suggest.enabled)`.
+
+By default, on request of completions `kb(workbench.action.terminal.requestCompletions)` files, folders, commands, arguments, and options are provided.
+
+![A user has typed git checkout in a terminal and requests completions. The completion list shows branch names like main and merogge/terminal-section](images/shell-integration/terminal-suggest.png)
+
+Commands are sourced from [Fig specs](https://github.com/withfig) and per-shell builtin functions (for `pwsh`, `bash`, `zsh`, and `fish`) and are validated against the `$PATH` to ensure they exist. On Windows, the particular set of executables can be configured with `setting(terminal.integrated.suggest.windowsExecutableExtensions)`.
+
+There can be multiple IntelliSense providers, which can be changed with `setting(terminal.integrated.suggest.providers)`.
+
+### Keyboard navigation
+
+By default, `Tab` inserts the suggestion. Once navigation of the list has occurred, `Enter` will similarly insert the suggestion. This can be configured with `setting(terminal.integrated.suggest.selectionMode)`.
```

Comment:
```suggestion
By default, `kbstyle(Tab)` inserts the suggestion. Once navigation of the list has occurred, `kbstyle(Enter)` will similarly insert the suggestion. You can configure this behavior with the `setting(terminal.integrated.suggest.selectionMode)` setting.
```

Reply:
Just edited for style but I don't really understand what "once navigation of the list has occurred" means. You might want to clarify.